### DISCUSSION TITLE
How does serve command process environment variables?

### DISCUSSION THREAD

**Discussion by ttc0419** [laravel/framework#40807] [Q&A]
According to https://github.com/laravel/framework/blob/09c1c0431c6cb616560f06e5740eb567364a96ab/src/Illuminate/Foundation/Console/ServeCommand.php#L96-L119
Only environment variables in the arrays will be set to original values, others will be set to false.  Why other environment variables are set to false?

And as a lot of people noticed, the command is not working properly on windows.  It will fail with "reason ?".  However, if I change the callback function to just ```return [$key => $value] ```, the serve command will run properly.  Why is that and can it be changed to just return the original key value pairs?

**Comment by fbsanchez** [comment]
Hello, custom .env file seems not to be working for me (Just created this issue but in the wrong place I suspect :sweat_smile: ) 

https://github.com/laravel/framework/issues/40978
